client/deposits/overview/instant-deposit-button.js

```diff
@@ -29,34 +66,68 @@ const InstantDepositModal = ( { amount, fee, net, onClose, onSubmit, inProgress
 					currency.formatCurrency( net / 100 )
 				) }
 			</Button>
+			<Button isDefault onClick={ onClose }>
+				{ __( 'Cancel', 'woocommerce-payments' ) }
+			</Button>
 		</Modal>
-	)
-}
+	);
+};
+
+const InstantDepositSubmitNotice = ( { error } ) => {
+	let message = '';
+	// TODO: supply proper error notices.
+	switch ( error.code ) {
+		case 'balance_insufficient':
+			message = __(
+				'Balance is not sufficient for instant deposit.',
+				'woocommerce-payments'
+			);
+			break;
+
+		default:
+			message = __(
+				'There was an error, please try again.',
+				'woocommerce-payments'
+			);
+	}
+
+	return (
+		<Notice status="error" isDismissible={ false }>
+			<p>
+				<strong>{ message }</strong>
+			</p>
+		</Notice>
+	);
+};
 
 // TODO: Properly style :allthethings:
-const InstantDepositButton = ( { balance: { amount, fee, net, transaction_ids } } ) => {
+const InstantDepositButton = ( {
+	balance: { amount, fee, net, transaction_ids: transactionIds },
+} ) => {
 	const [ isModalOpen, setModalOpen ] = useState( false );
 	const [ inProgress, setInProgress ] = useState( false );
+	const [ hasError, setHasError ] = useState( false );
 
 	// TODO: Use wp.data
 	const submit = async () => {
 		try {
 			setInProgress( true );
+			setHasError( false );
 			await apiFetch( {
 				path: '/wc/v3/payments/deposits',
 				method: 'POST',
 				data: {
 					type: 'instant',
-					transaction_ids,
+					transactionIds,
 				},
 			} );
 			// TODO: Success notice? Full-reload the page so the new deposit appears?
```

Comment:
Related to the error UI comment. What do you think we should do on success? Success notice in the modal? Close the modal and show a success notice in the page? We will also have to somehow re-fetch the data from the server so that the overview amounts and the list of deposits are correct.

Reply:
It might be easier to show the success notice in the modal, and when they close it, then refresh the page. This would be instead of us closing the modal, refreshing the page, and then showing the notice.

Reply:
Sounds good 👍